File: 6/StringCost.java
class StringCost {

  public static void main(String[] args) {

    if (args.length < 2) {
      System.err.println("usage : java StringCost [type: plus or buffer] [count]");
      System.exit(1);
    }
    int count = Integer.parseInt(args[1]);

    long before = System.currentTimeMillis();
    if ("plus".equals(args[0])) {

      String buffer = "";
      for (int i = 0; i < count; i++) {
        buffer += "abc";
      }

    } else if ("buffer".equals(args[0])) {

      StringBuilder builder = new StringBuilder();
      for (int i = 0; i < count; i++) {
        builder.append("abc");
      }
    }
 
    System.out.println(String.format("%d ms", System.currentTimeMillis() - before));
    System.exit(0);
  }
}

File: 6/string_cost.rb
#!/usr/bin/ruby 

if (ARGV.length < 2)
  puts("usage : ruby string_cost.rb [type: plus or buffer] [count]")
  exit(1)
end

count = ARGV[1].to_i

before = Time.now
String buffer = ""
if ("plus" == ARGV[0]) 

  (0..count).each do | i |
    buffer += "abc"
  end

elsif ("buffer" == ARGV[0])

  (0..count).each do | i |
    buffer << "abc"
  end
end

puts "#{(Time.now - before) * 1000} ms"
exit(0)